pkg/server/test/write_assertions.go

```diff
@@ -208,3 +250,134 @@ type repo
+		_name      string
+		assertions []*openfgav1.Assertion
+		modelID    string
+		err        error
+	}
+
+	store := testutils.CreateRandomString(10)
+
+	githubModelReq := &openfgav1.WriteAuthorizationModelRequest{
+		StoreId: store,
+		TypeDefinitions: parser.MustTransformDSLToProto(`model
+	schema 1.1
+type user
+
+type repo
+  relations
+	define reader: [user]
+	define can_read: reader`).GetTypeDefinitions(),
+		SchemaVersion: typesystem.SchemaVersion1_1,
+	}
+	ctx := context.Background()
+
+	writeAuthzModelCmd := commands.NewWriteAuthorizationModelCommand(datastore)
+	modelID, err := writeAuthzModelCmd.Execute(ctx, githubModelReq)
+	require.NoError(t, err)
+
+	var tests = []writeAssertionsTestSettings{
+		{
+			_name: "write_conceptual_tuple_assertion_with_invalid_relation_fails",
+			assertions: []*openfgav1.Assertion{
+				{
+					ContextualTuples: &openfgav1.ContextualTupleKeys{
+						TupleKeys: []*openfgav1.TupleKey{
+							{
+								User:     "user:jon",
+								Object:   "repo:test",
+								Relation: "invalidrelation",
+							},
+						},
+					},
+					Expectation: false,
+				},
+			},
+			modelID: modelID.GetAuthorizationModelId(),
+			err: serverErrors.ValidationError(
+				fmt.Errorf("relation 'repo#invalidrelation' not found"),
+			),
+		},
+		{
+			_name: "write_conceptual_tuple_assertion_with_invalid_object_fails",
+			assertions: []*openfgav1.Assertion{
+				{
+					ContextualTuples: &openfgav1.ContextualTupleKeys{
+						TupleKeys: []*openfgav1.TupleKey{
+							{
+								User:     "user:jon",
+								Object:   "invalidobect",
+								Relation: "can_read",
+							},
+						},
+					},
+					Expectation: false,
+				},
+			},
+			modelID: modelID.GetAuthorizationModelId(),
+			err: serverErrors.ValidationError(
+				fmt.Errorf("invalid 'object' field format"),
+			),
+		},
+		{
+			_name: "write_conceptual_tuple_assertion_with_invalid_object_fails",
+			assertions: []*openfgav1.Assertion{
+				{
+					ContextualTuples: &openfgav1.ContextualTupleKeys{
+						TupleKeys: []*openfgav1.TupleKey{
+							{
+								User:     "invaliduser",
+								Object:   "repo:test",
+								Relation: "can_read",
+							},
+						},
+					},
+					Expectation: false,
+				},
+			},
+			modelID: modelID.GetAuthorizationModelId(),
+			err: serverErrors.ValidationError(
+				fmt.Errorf("the 'user' field must be an object (e.g. document:1) or an 'object#relation' or a typed wildcard (e.g. group:*)"),
+			),
+		},
+
+		{
+			_name: "writing_contextual_tuples_assertion_with_not_found_id",
+			assertions: []*openfgav1.Assertion{
+				{
+					ContextualTuples: &openfgav1.ContextualTupleKeys{
+						TupleKeys: []*openfgav1.TupleKey{
+							{
+								User:     "user:smeadows",
+								Object:   "repo:test",
+								Relation: "can_read",
+							},
+						},
+					},
+					Expectation: false,
+				},
+			},
+			modelID: "not_valid_id",
+			err: serverErrors.AuthorizationModelNotFound(
+				"not_valid_id",
+			),
+		},
```

Comment:
This test isn't useful - validation of contextual happens will never happen if the model ID isn't valid.

Reply:
👍